source/reference/array.rst

```diff
@@ -236,7 +236,200 @@ Unevaluated Items
+                "$schema": "https://json-schema.org/draft/2020-12/schema",
+                "if": {
+                    "prefixItems": [{"const": "a"}]
+                },
+                "unevaluatedItems": false
+            },
+            "tests": [
+                {
+                    "description": "valid in case if is evaluated",
+                    "data": [ "a" ],
+                    "valid": true
+                },
+                {
+                    "description": "invalid in case if is evaluated",
+                    "data": [ "b" ],
+                    "valid": false
+                }
+            ]
+    }
+
+And an important note: ``unevaluatedItems`` can't see inside cousins
+(a vertically adjacent item inside a separate pair of {curly braces}
+with the same "parent"— ``anyOf``, ``if``, ``not``, or similar). Such
+an instance always fails evaluation.
+
+.. schema_example::
+
+    {
+        "description": "unevaluatedItems can't see inside cousins",
+        "schema": {
+            "$schema": "https://json-schema.org/draft/2020-12/schema",
+            "allOf": [
+                {
+                  "prefixItems": [ true ]
+                },
+                { "unevaluatedItems": false }
+            ]
+        },
+        "tests": [
+            {
+                "description": "always fails",
+                "data": [ 1 ],
+                "valid": false
+            }
+        ]
+    }
+
+Finally, here's an example of ``unevaluatedItems`` if you're
+`structuring`. Let's make a "half-closed" schema: something useful
+when you want to keep the first two arguments, but also add more in
+certain situations. ("Closed" to two in some places, "open" to more
+in others.)
+
+.. schema_example::
+
+    {
+      "$id": "https://example.com/my-tuple",
+
+      "type": "array",
+      "prefixItems": [
+        true,
+        { "type": "boolean" }
+      ],
+
+      "$defs": {
+        "closed": {
+          "$anchor": "closed",
+          "$ref": "#",
+          "unevaluatedItems": false
+        }
+      }
+    }
+
+Then we can extend the tuple:
+
+.. schema_example::
+
+    {
+      "$id": "https://example.com/my-extended-tuple",
+
+      "$ref": "/my-tuple",
+      "prefixItems": [
+        true,
+        true,
+        { "type": "boolean" }
+      ],
+      "unevaluatedItems": false
+    }
+
+With this, you can use ``$ref`` to reference the first two
+``prefixItems`` and keep the schema "closed" to two when necessary,
+"open" to more when necessary. A reference to ``/my-tuple#closed``
+would disallow more than two items, when you need it to.
+
+.. note::
+   For a tower of more examples, read our `unevaluatedItems Test Suite <https://github.com/json-schema-org/JSON-Schema-Test-Suite/blob/main/tests/draft2020-12/unevaluatedItems.json>`_ on GitHub.
+   We test a lot of use cases there, including uncommon ones. Do any
+   of these apply to your schema?
+   - using ``unevaluatedItems`` as a schema
```

Comment:
I assume these were TODO notes to yourself perhaps from here down and meant to be removed?

Reply:
I tried to explain the base concepts of ``unevaluatedItems`` on this page and give jumping-off points for other cases readers might have. While I could have explained all the cases in the test suite, I felt five schema examples for one subheader was getting bloated. So I whittled down what I thought were the most abstract, general cases that needed explanation, then said essentially "Through this intro, I hope I've given you the tools to click to the test suite and understand some of the more uncommon, harder use cases, if they apply to you."

Reply:
Ohhh I see, the bulleted list was meant to be the thing referred to by "Do any of these apply to your schema?" -- I have to say that wasn't even clear to me -- I though that sentence was trying to say "those" and simply saying "go check the test suite and see if any further examples apply to you".

I'm certainly supportive of not defining all the examples in this doc! Though I think we should then massage the bullets here, I have to reread them to see whether I can visualize which cases they're talking about. Thanks for the clarification.